src/react/useLocations.ts

```diff
@@ -0,0 +1,72 @@
+import { useContext, useEffect, useRef } from 'react';
+import { SpaceContext } from './contexts/SpaceContext.js';
+import { isArray, isFunction, isString } from '../utilities/is.js';
+
+import type Locations from '../Locations.js';
+import type { SpaceMember } from '../types.js';
+import type { UseSpaceOptions } from './types.js';
+
+interface UseLocationsResult {
+  update: Locations['set'];
+}
+
+type UseLocationCallback = (locationUpdate: { member: SpaceMember }) => void;
+
+export type LocationsEvent = 'update';
+
+function useLocations(callback?: UseLocationCallback, options?: UseSpaceOptions): UseLocationsResult;
+function useLocations(
+  event: LocationsEvent | LocationsEvent[],
+  callback: UseLocationCallback,
+  options?: UseSpaceOptions,
+): UseLocationsResult;
+
+/*
+ * Registers a subscription on the `Space.locations` object
+ */
+export function useLocations(
+  eventOrCallback: LocationsEvent | LocationsEvent[] | UseLocationCallback,
+  callbackOrOptions?: UseLocationCallback | UseSpaceOptions,
+  optionsOrNothing?: UseSpaceOptions,
+): UseLocationsResult {
+  const space = useContext(SpaceContext);
```

Comment:
I wonder if the `SpaceContext` should hold connection and channel errors? This way we can wrap functions like `update` and have them not be called when connections are not available? WDYT?

Reply:
Sounds like a good I idea!